src/FluentResults/Results/Result.cs

```diff
@@ -148,6 +148,56 @@ public Result<TNewValue> ToResult<TNewValue>(Func<TValue, TNewValue> valueConver
                 .WithReasons(Reasons);
         }
 
+        /// <summary>
+        /// Convert result with value to result with another value that may fail.
+        /// </summary>
+        /// <example>
+        /// <code>
+        ///  var bakeryDtoResult = result
+        ///     .Bind(GetWhichMayFail)
+        ///     .Bind(ProcessWhichMayFail)
+        ///     .Bind(FormattingWhichMayFail);
+        /// </code>
+        /// </example>
+        /// <param name="valueConverter">Transformation that may fail.</param>
+        public Result<TNewValue> Bind<TNewValue>(Func<TValue, Result<TNewValue>> valueConverter)
+        {
+            var result = new Result<TNewValue>();
+            result.WithReasons(Reasons);
+            
+            if (IsSuccess)
+            {
+                var converted = valueConverter(Value);
+                result.WithValue(converted.ValueOrDefault);
+                result.WithReasons(converted.Reasons);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Convert result with value to <see cref="Result"/>.
+        /// </summary>
+        /// <example>
+        /// <code>
+        ///  var done = result.Bind(ActionWhichMayFail);
+        /// </code>
+        /// </example>
+        /// <param name="action">Action that may fail.</param>
+        public Result Bind(Func<TValue, Result> action)
```

Comment:
I think it would be good if you can add also a async/await version of it.

Reply:
Last week we added ValueTask support to FluentResults. Please also provide a ValueTask version of the methods.

Reply:
I agree and planning to talk about it after merging this one in a new issue. However, I will implement it as part of this change. `BindAsync` will be. It should work both both `Task` and `ValueTask`.

Reply:
I added overloads to `Bind` instead of `BindAsync`